ksql-engine/src/main/java/io/confluent/ksql/query/BlockingQueryQueue.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query;
+
+import io.confluent.ksql.GenericRow;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.streams.KeyValue;
+
+/**
+ * The queue between the Kafka-streams topology and the client connection.
+ *
+ * <p>The KS topology writes to the queue from its {@code StreamThread}, while the KSQL server
+ * thread that is servicing the client request reads from the queue and writes to the client
+ * socket.
+ */
+public interface BlockingQueryQueue {
+
+  void setLimitHandler(LimitHandler limitHandler);
```

Comment:
would be nice to also document this (or add documentation to `LimitHandler`) to describe when exactly the limit is considered reached

Reply:
sure